core/unix/memquery.c

```diff
@@ -144,6 +145,9 @@ memquery_library_bounds_by_iterator(const char *name, app_pc *start /*IN/OUT*/,
                         /* we keep the last '/' at end */
                         ++slash;
                         strncpy(dst, src, MIN(dstsz, (slash - src)));
+                        if (filename != NULL)
```

Comment:
`&& slash != NULL`?  Else change the `slash != NULL` to an invariant by removing `_CURIOSITY`.

Reply:
Done